app/controllers/admin/announcements_controller.rb

```diff
@@ -32,8 +32,12 @@ def update
   end
 
   def destroy
-    redirect_to admin_announcements_path,
-                success: t('.success', title: @announcement.title) if @announcement.destroy
+    if @announcement.destroy
+      redirect_to admin_announcements_path,
+                  success: t('.success', title: @announcement.title)
+    else
+      redirect_to admin_announcements_path, danger: t('.failure')
```

Comment:
Flash data is persisted in a cookie, you don't have much options other than text.